docs/DeveloperGuide.md

```diff
@@ -1757,22 +1757,158 @@ testers are expected to do more *exploratory* testing.
+      
+   1. Test case: `delete -1`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+   1. Test case: `delete x`, where x > number of students in the list<br>
+      Expected: No student is deleted. Error details shown in the status message.
 
-   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
-      Expected: Similar to previous.
+1. Deleting a student using their student ID
 
-1. _{ more test cases …​ }_
+      1. Prerequisites: 
+          1. Sample data is loaded in TAPA. 
+          2. One student has "A0000000Z" as their student ID. 
+          3. No student has "A9999999X" as their student ID. 
+
+   1. Test case: `delete i/A0000000Z`<br>
+      Expected: The student with student ID "A0000000Z" deleted from the list. Details of the deleted student shown in the status message.
+      
+   1. Test case: `delete i/A9999999X`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+
+   1. Test case: `delete i/@@@`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+   1. Test case: `delete i/`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+   1. Test case: `delete`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+### Deleting students by module code
+
+1. Deleting students by module code
+
+   1. Prerequisites: 
+         1. Sample data is loaded in TAPA. 
+         2. One or more students have "CS2100" as their module code.
+         3. No students have "CS9999" as their module code.
+
+   1. Test case: `deleteModule m/CS2100`<br>
+      Expected: The students with module code "CS2100" are deleted from the list. Status message states number of students deleted.
+      
+   1. Test case: `deleteModule m/CS9999`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+
+   1. Test case: `deleteModule m/@@@`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+    1. Test case: `deleteModule m/`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+    1. Test case: `deleteModule`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+    1. Test case: `deleteModule m/CS CS`<br>
+      Expected: No student is deleted. Error details shown in the status message.
+      
+### Finding students
+
+1. Finding a student using their name
+
+      1. Prerequisites: 
+          1. Sample data is loaded in TAPA. 
+          2. One or more students have "John" in their name.
+          3. No students have "Dueet" in their name.
+
+   1. Test case: `find n/John`<br>
+      Expected: All students with "John" in their names are listed. Status message states how many students were listed.
```

Comment:
I'm not sure if this phrasing could be slightly misleading since `find n/John` will not return Johnathan.
Perhaps this might be clearer? 

```suggestion
      Expected: All students with "John" as their names are listed. Status message states how many students were listed.
```

Reply:
Thanks for the review @brelkh! Regarding this, I wrote ""John" in their names" as the student's name might not be just John, but rather John Doe or John Tan etc., whereas "as their names" somewhat implies that their entire name is just John. But I definitely do see where you're coming from. I might add a clarification in this segment.